service/history/mutableStateBuilder.go

```diff
@@ -324,7 +324,9 @@ func (e *mutableStateBuilder) updateReplicationStateVersion(version int64) {
 
 // Assumption: It is expected CurrentVersion on replication state is updated at the start of transaction when
 // mutableState is loaded for this workflow execution.
-func (e *mutableStateBuilder) updateReplicationStateLastEventID(clusterName string, lastEventID int64) {
+func (e *mutableStateBuilder) updateReplicationStateLastEventID(clusterName string, lastWriteVersion,
+	lastEventID int64) {
+	e.replicationState.LastWriteVersion = lastWriteVersion
 	e.replicationState.LastWriteVersion = e.replicationState.CurrentVersion
```

Comment:
this line should be deleted.

Reply:
and the active workflow, when calling this function, should use the current version

Reply:
fixed.